modules/imgproc/src/median_blur.simd.hpp

```diff
@@ -489,7 +490,149 @@ medianBlur_8u_Om( const Mat& _src, Mat& _dst, int m )
 #undef N
 #undef UPDATE_ACC
 }
+#endif
+
+// Binary search to find the median offset in the histogram
+static inline void
+get_median_ofs256(int &ofs, const uint16_t *hist, const int halfsum) {
+    int s = 0, step = 128;
+    uint16_t ds, m;
 
+#if CV_SIMD || CV_SIMD_SCALABLE
+    const int lanes = VTraits<v_uint16>::vlanes();
+#endif
+
+    ofs = 0;
+    while (step) {
+#if CV_SIMD || CV_SIMD_SCALABLE
+        // Use SIMD instructions when the step is larger than or equal to the lane size
+        if (step >= lanes) {
+            v_uint16 v_ds = vx_load(hist + ofs);
+            for (int i = lanes; i <= step - lanes; i += lanes)
+                v_ds = v_add(v_ds, vx_load(hist + ofs + i));
+            ds = v_reduce_sum(v_ds);
+        }
+            // For smaller steps, use scalar accumulation
+        else
+#endif
+        {
+            ds = hist[ofs];
+            for (int i = 1; i < step; i++)
+                ds += hist[ofs + i];
+        }
+
+        // Determine if the cumulative sum has reached or surpassed half the total sum
+        m = (s + ds) <= halfsum;
+        ofs += m * step;
+        s += ds & -m;
+        step = step >> 1;
+    }
+}
+
+static void
+medianBlur_8u(const Mat &_src, Mat &_dst, int ksize) {
+    CV_INSTRUMENT_REGION();
+
+    const int channels = _src.channels();
+    int radius = ksize / 2;
+    CV_Assert(ksize % 2 == 1);
+
+    const int rows = _src.rows;
+    const int cols = _src.cols;
+    const int win_half_size = ksize * ksize / 2;
+
+    const uint8_t *src_data = _src.ptr<uint8_t>();
+    size_t src_step = _src.step;
+    uint8_t *dst_data = _dst.ptr<uint8_t>();
+    size_t dst_step = _dst.step;
+
+    double nstripes = (double) (rows * cols) / 1024.0;
+    parallel_for_(Range(0, rows), [&](const Range &r) {
+        int y_start = r.start;
+        int y_end = r.end;
+
+        std::vector<std::vector<uint16_t>> hist256(channels, std::vector<uint16_t>(256, 0));
+        std::vector<const uint8_t *> win_row_ptrs(ksize);
```

Comment:
extract pointer from hist256 and win_row_ptrs. Accessing `std::vector` may be significantly slower than accessing plain arrays.

```
std::vector<...> hist256vec(...);
uint16_t* hist256 = hist256vec.data();
std::vector<...> win_row_ptrs_vec(...);
const uint8_t* win_row_ptrs = win_row_ptrs_vec.data();
```